pkg/apis/apps/validation/validation.go

```diff
@@ -141,3 +141,35 @@ func ValidateStatefulSetStatusUpdate(statefulSet, oldStatefulSet *apps.StatefulS
 	// TODO: Validate status.
 	return allErrs
 }
+
+// ValidateControllerRevisionName can be used to check whether the given ControllerRevision name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+var ValidateControllerRevisionName = apivalidation.NameIsDNSSubdomain
+
+// ValidateControllerRevision collects errors for the fields of state and returns those errors as an ErrorList. If the
+// returned list is empty, state is valid. Validation is performed to ensure that state is a valid ObjectMeta, its name
+// is valid, and that it doesn't exceed the MaxControllerRevisionSize.
+func ValidateControllerRevision(revision *apps.ControllerRevision) field.ErrorList {
+	errs := field.ErrorList{}
+
+	errs = append(errs, apivalidation.ValidateObjectMeta(&revision.ObjectMeta, true, ValidateControllerRevisionName, field.NewPath("metadata"))...)
+	if revision.Data == nil {
+		errs = append(errs, field.Required(field.NewPath("data"), "data is mandatory"))
+	}
+	errs = append(errs, apivalidation.ValidateNonnegativeField(revision.Revision, field.NewPath("revision"))...)
```

Comment:
I think that a ControllerRevision with a `.Revision` of 0 is a valid object. It doesn't enforce that the end user ever creates such an object, but implementing validation based on a convention for kubectl flag default seems to be mixing concerns imo. What if the flag is defaulted to -1 when non-present in the future, or if a nil pointer is used to indicate its absence. I don't see a strong reason to say that a ControllerRevision with `.Revision` == 0, is not a valid object.